csw-server/src/main/java/org/fao/geonet/component/csw/Transaction.java

```diff
@@ -527,20 +544,61 @@ private void getResponseResult(Element request, Element response, List<String> f
             Element insertResult = new Element("InsertResult", Csw.NAMESPACE_CSW);
             insertResult.setAttribute("handleRef", "handleRefValue");
 
-            //Namespace NAMESPACE_DC = Namespace.getNamespace("dc", "http://purl.org/dc/elements/1.1/");
-            for (String fileId : fileIds) {
-                Element briefRecord = new Element("BriefRecord", Csw.NAMESPACE_CSW);
-                Element identifier = new Element("identifier"); //,Csw.NAMESPACE_DC );
-                identifier.setText(fileId);
-                briefRecord.addContent(identifier);
+            for (InsertedMetadata md : inserted) {
+                Element briefRecord;
+
+                try {
+                    briefRecord = applyCswBrief(context, _schemaManager, md.schema, md.content, md.id, context.getLanguage());
+
+                } catch(Exception e) {
+                    // let's mock a BriefRecord, it's not complete, but it may be useful anyway
+                    Log.warning(Geonet.CSW, "Error transforming metadata: " + md.id, e);
+
+                    briefRecord = new Element("BriefRecord", Csw.NAMESPACE_CSW);
+                    Element identifier = new Element("identifier", Csw.NAMESPACE_DC );
+                    // TODO: title is mandatory
+                    identifier.setText(md.id);
+                    briefRecord.addContent(identifier);
+                }
                 insertResult.addContent(briefRecord);
             }
 
             response.addContent(insertResult);
         }
+    }
 
+
+    private static Element applyCswBrief(ServiceContext context, SchemaManager schemaManager, String schema,
+                                               Element md,
+                                               String id, String displayLanguage) throws Exception
+    {
+        return applyElementSetName(context, schemaManager, schema, md, "csw", ElementSetName.BRIEF, id, displayLanguage);
     }
 
+    private static Element applyElementSetName(ServiceContext context, SchemaManager smgr, String schema,
```

Comment:
We can not use this https://github.com/geonetwork/core-geonetwork/blob/master/csw-server/src/main/java/org/fao/geonet/kernel/csw/services/getrecords/SearchController.java#L178 which is similar ?

Reply:
Yes sure. I copied the code from another class, that was not in the dependencies for this module.